### PR TITLE
Optional email configs

### PR DESCRIPTION
**Pull request by ocervell**
- Add flag `email_notifications_enabled` to leave out email configuration for notifications (if needed).
- Add input checks to fail early
- Add details to README on CSCC integration
- Add email config to `simple_example`

### CODE REVIEW

**Comment by ocervell** [comment]
Fix #161

**Comment by ocervell** [comment]
@aaron-lane I added the `email_notifications_enabled` flag because there is plan for Forseti to add more email services (see [here](https://github.com/forseti-security/forseti-security/issues/859).

**Comment by ocervell** [comment]
@aaron-lane removed the flag, implemented your suggestion

File: examples/simple_example/variables.tf [hunk 1]
```diff
@@ -50,3 +50,18 @@ variable "private" {
   description = "Private client and server instances (no public IPs)"
   default     = true
 }
+
+variable "sendgrid_api_key" {
+  description = "Sendgrid API key (if `email_violations_enabled` is set)"
```
Comment (aaron-lane) [inline examples/simple_example/variables.tf]:
```suggestion
  description = "Sendgrid API key."
```

File: examples/simple_example/variables.tf [hunk 2]
```diff
@@ -50,3 +50,18 @@ variable "private" {
...
+  default     = ""
+}
+
+variable "forseti_email_sender" {
+  description = "Forseti email sender (if `email_violations_enabled` is set)"
```
Comment (aaron-lane) [inline examples/simple_example/variables.tf]:
```suggestion
  description = "Forseti email sender."
```

File: examples/simple_example/variables.tf [hunk 3]
```diff
@@ -50,3 +50,18 @@ variable "private" {
...
+  default     = ""
+}
+
+variable "forseti_email_recipient" {
+  description = "Forseti email recipient (if `email_violations_enabled` is set)"
```
Comment (aaron-lane) [inline examples/simple_example/variables.tf]:
```suggestion
  description = "Forseti email recipient."
```

**Comment by ocervell** [comment]
@aaron-lane Applied your changes, rebased against master, updated Changelog

**Comment by morgante** [comment]
I'm going ahead and merging this since the CI failure looks unrelated.

**Pull request merged**